src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -306,6 +307,30 @@ public void AddHostObject(string name, FormulaType type, Func<IServiceProvider,
                 displayName: default);
 
             _variables.Add(hostDName, info);
+        }
+
+        /// <summary>
+        /// Adds a UserInfo object schema.
+        /// Actual object is added in Runtime config service provider.
+        /// </summary>
+        public void AddUserInfoObject()
+        {
+            var userInfoType = RecordType.Empty()
+                .Add("FullName", FormulaType.String)
+                .Add("Email", FormulaType.String);
+
+            AddHostObject("UserInfo", userInfoType, GetUserInfoObject);
+        }
+
+        private FormulaValue GetUserInfoObject(IServiceProvider serviceProvider)
+        {
+            var userInfo = (IUserInfo)serviceProvider.GetService(typeof(IUserInfo)) ?? throw new Exception("UserInfo object was not added to service");
+
+            RecordValue userRecord = FormulaValue.NewRecordFromFields(
+                new NamedValue("FullName", FormulaValue.New(userInfo.FullName ?? string.Empty)),
+                new NamedValue("Email", FormulaValue.New(userInfo.Email ?? string.Empty)));
+
+            return userRecord;
         }
```

Comment:
Can we move these to an extension method, and not directly on SymbolTable.

Reply:
yes, moved this to extension method,